tests/test_concurrency/test_immediatescheduler.py

```diff
@@ -43,10 +45,24 @@ class MyException(Exception):
         def action(scheduler, state=None):
             raise MyException()
 
-        try:
+        with pytest.raises(MyException):
             return scheduler.schedule(action)
-        except MyException:
-            assert True
+
+    def test_immediate_schedule_action_due_error(self):
+        scheduler = ImmediateScheduler()
+        ran = False
+
+        class MyException(Exception):
```

Comment:
I believe the local class `MyException` is unnecessary here?

Reply:
Yes, yes it is. I will remove it, thanks!